flake.nix

```diff
@@ -0,0 +1,221 @@
+{
+  description = "BDK Flake to run all tests locally and in CI";
+
+  inputs = {
+    # stable nixpkgs (let's not YOLO on unstable)
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
+
+    # pin dependencies to a specific version
+    # find instructions here:
+    # <https://lazamar.co.uk/nix-versions>
+
+    # bitcoind pinned to 0.25.1
+    nixpkgs-bitcoind.url = "github:nixos/nixpkgs?rev=53793ca0aecf67164c630ca34dbfde552a8ab085";
```

Comment:
There are always backports on bitcoin `25.X`.  The latest one was three weeks ago.  The current way to pin bitcoin `0.25.1` is not updated because it is based on the date `2023-10-19`.  And pin the all `nixos/nixpkgs` is pretty heavy.

May we should build bitcoind with correct version?

Reply:
> May we pin bitcoind with a correct version with hash but not nixos/nixpkgs?

What do you mean by that?
Downloading the binary straight from `https://bitcoin.org/`?

Reply:
@evanlinjin If it is good to put it into another issue and keep this PR as simple as possible.  I am happy to be assigned to the other one.

Reply:
Oh noice! Yes, a full derivation. Now I get the idea!